lib/rules/no-extra-boolean-cast.js

```diff
@@ -174,6 +192,7 @@ module.exports = {
                 case "DoWhileStatement":
                 case "WhileStatement":
                 case "ForStatement":
+                case "SequenceExpression":
                     return false;
                 case "ConditionalExpression":
                     return precedence(node) <= precedence(parent);
```

Comment:
This was assuming that `previousNode` is `parent.test` (as in the description of this function on line 167, which we should also update), but now it can be `consequent`/`alternate`, in which case only `SequenceExpression` would need parentheses.

For example:

```js
/* eslint no-extra-boolean-cast: 2 */

if (a ? Boolean(b = c) : Boolean(d = e));

```

After the autofix:

```js
/* eslint no-extra-boolean-cast: 2 */

if (a ? (b = c) : (d = e));

```

Parentheses around `b = c` and `d = e` are unnecessary.

Reply:
This one took me a while to figure out trying to make use of the existing infrastructure. 
I would have expected to be able to write 
```js
                    if (previousNode === parent.test) {
                        return precedence(node) <= precedence(parent);
                    }
                    if (previousNode === parent.consequent || previousNode === parent.alternate) {
                        return precedence(node) < precedence(parent);
                    }
```
but it seems that eslint's precedence() function differs from MDN's table, in that it treats AssignmentExpressions and similar as lower precedence than ternary, rather than equal... See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence#table, vs

https://github.com/eslint/eslint/blob/239a7e27209a6b861d634b3ef245ebbb805793a3/lib/rules/utils/ast-utils.js#L1509-L1602

Think that should be a followup to change to match the MDN version, or is it better to just leave the current system?

Reply:
We can consider changing this in a followup if the current logic is confusing and the new one would be better for practical use, but we'll need to check all rules that use `getPrecedence()` to see if they need to be updated.